htdocs/compta/facture/class/facture-rec.class.php

```diff
@@ -1058,44 +1071,44 @@ function createRecurringInvoices($restictoninvoiceid=0, $forcevalidation=0)
 
 					dol_syslog("createRecurringInvoices Process invoice template id=".$facturerec->id.", ref=".$facturerec->ref.", entity=".$facturerec->entity);
 
-				    $facture = new Facture($db);
+					$facture = new Facture($db);
 					$facture->fac_rec = $facturerec->id;    // We will create $facture from this recurring invoice
 					$facture->fk_fac_rec_source = $facturerec->id;    // We will create $facture from this recurring invoice
 
-				    $facture->type = self::TYPE_STANDARD;
-				    $facture->brouillon = 1;
-				    $facture->date = (empty($facturerec->date_when)?$now:$facturerec->date_when);	// We could also use dol_now here but we prefer date_when so invoice has real date when we would like even if we generate later.
-				    $facture->socid = $facturerec->socid;
-
-				    $invoiceidgenerated = $facture->create($user);
-				    if ($invoiceidgenerated <= 0)
-				    {
-				        $this->errors = $facture->errors;
-				        $this->error = $facture->error;
-				        $error++;
-				    }
-				    if (! $error && ($facturerec->auto_validate || $forcevalidation))
-				    {
-				        $result = $facture->validate($user);
-				        if ($result <= 0)
-				        {
-	    			        $this->errors = $facture->errors;
-	    			        $this->error = $facture->error;
-				            $error++;
-	                    }
-				    }
-	                if (! $error && $facturerec->generate_pdf)
-	                {
-	                    // We refresh the object in order to have all necessary data (like date_lim_reglement)
-	                    $facture->fetch($facture->id);
-	                    $result = $facture->generateDocument($facturerec->modelpdf, $langs);
-	                    if ($result <= 0)
-	                    {
-	                        $this->errors = $facture->errors;
-	                        $this->error = $facture->error;
-	                        $error++;
-	                    }
-	                }
+					$facture->type = self::TYPE_STANDARD;
+					$facture->brouillon = 1;
+					$facture->date = (empty($facturerec->date_when)?$now:$facturerec->date_when);	// We could also use dol_now here but we prefer date_when so invoice has real date when we would like even if we generate later.
+					$facture->socid = $facturerec->socid;
+
+					$invoiceidgenerated = $facture->create($user);
+					if ($invoiceidgenerated <= 0)
+					{
+						$this->errors[] = $facture->errors;
```

Comment:
@bugness-chl 
$facture->errors is already an array !
$this->errors = $facture->errors is correct